File: 2021/05/12/main.go
package main

type NumMatrix struct {
	mat [][]int
}

func Constructor(matrix [][]int) NumMatrix {
	return NumMatrix{mat: matrix}
}

func (this *NumMatrix) SumRegion(row1 int, col1 int, row2 int, col2 int) int {
	s := 0
	for i := row1; i <= row2; i++ {
		for j := col1; j <= col2; j++ {
			s += this.mat[i][j]
		}
	}
	return s
}

/**
 * Your NumMatrix object will be instantiated and called as such:
 * obj := Constructor(matrix);
 * param_1 := obj.SumRegion(row1,col1,row2,col2);
 */

File: 2021/05/12/main.py
class NumMatrix:

    def __init__(self, matrix: List[List[int]]):
        self.mat = matrix

    def sumRegion(self, row1: int, col1: int, row2: int, col2: int) -> int:
        s = 0
        for r in range(row1, row2+1):
            s += self.mat[r][col1:col2+1]
        return s


# Your NumMatrix object will be instantiated and called as such:
# obj = NumMatrix(matrix)
# param_1 = obj.sumRegion(row1,col1,row2,col2)
